brigade.js

```diff
@@ -42,7 +44,61 @@ function build(e, project) {
   ];
 
   // Run in parallel
-  Group.runAll([jsTest, goBuild])
+  Group.runAll([jsTest, goBuild]).then( () => {
+    const gh = JSON.parse(e.payload)
+    if (e.event == "push" && gh.ref.startsWith("refs/tags/")) {
+      release(e, p)
+    }
+  })
+}
+
+function release(e, p) {
+  if (!p.secrets.ghToken) {
+    throw new Error("Project must have 'secrets.ghToken' set")
+  }
+
+  const tag = e.commit
+  const binName = "brig"
+  const gopath = "/go"
+  const localPath = gopath + "/src/github.com/" + p.repo.name;
+
+  // Cross-compile binaries for a given release and upload them to GitHub.
+  var cx = new Job("cross-compile", goImg)
+  cx.storage.enabled = true
+
+  parts = p.repo.name.split("/", 2)
+
+  cx.env = {
+    GITHUB_USER: parts[0],
+    GITHUB_REPO: parts[1],
+    GITHUB_TOKEN: p.secrets.ghToken,
+    GOPATH: gopath
+  }
+
+  cx.tasks = [
+    "go get github.com/golang/dep/cmd/dep",
+    "go get github.com/aktau/github-release",
+    // Need to move the source into GOPATH so vendor/ works as desired.
+    "mkdir -p " + localPath,
+    "cp -a /src/* " + localPath,
+    "cp -a /src/.git " + localPath,
+    "cd " + localPath,
+    "ls -lh",
+    "dep ensure",
+    "make build-release",
+    `github-release release -t ${tag} -n "${parts[1]} ${tag}"`
+  ]
+
+  // Upload for each target that we support
+  for (const f of ["linux-amd64", "windows-amd64", "darwin-amd64"]) {
+    const name = binName + "-"+f
+    cx.tasks.push(`github-release upload -R -f ./bin/${name} -n ${name} -t ${tag}`)
```

Comment:
Could we extract this release-upload sequence to e.g. `make publish-release`, just for the sake of consistency with `make build-release` and a bit more testability(it will be testable in isolation with brigade)?

Reply:
Or If you've intended to parallelize uploads with `Group.runAll` in the future, I believe we'd better leave this as-is for now.

Reply:
Nit: As `github-release release` fails when the release already exists, foreseeing uploads to replace existing ones by `github-release upload -R` seemed unnecessary. I believe `-R` can be safely omitted.

Reply:
Fixed.